voyager-core/src/androidMain/kotlin/cafe/adriel/voyager/androidx/AndroidScreenLifecycleOwner.kt

```diff
@@ -128,93 +37,26 @@ public class AndroidScreenLifecycleOwner private constructor() :
     }
 
     override fun onDispose(screen: Screen) {
-        val context = atomicContext.getAndSet(null) ?: return
-        if (context is Activity && context.isChangingConfigurations) return
-        viewModelStore.clear()
-        disposeEvents.forEach {
-            lifecycle.handleLifecycleEvent(it)
-        }
-    }
-
-    private fun performSave(outState: Bundle) {
-        controller.performSave(outState)
+        voyagerAndroidLifecycleOwner?.dispose()
+        voyagerAndroidLifecycleOwner = null
     }
 
     @Composable
     private fun getHooks(): List<ProvidedValue<*>> {
-        atomicContext.compareAndSet(null, LocalContext.current)
+        val androidLifecycleOwner = checkNotNull(voyagerAndroidLifecycleOwner) {
+            "ProvideBeforeScreenContent was not called before getHooks"
+        }
```

Comment:
This check here is really required? `getHooks` is a private function, we can just pass the `voyagerAndroidLifecycleOwner` to it.